github/downloader_test.go

```diff
@@ -80,74 +113,377 @@ func getDownloader() (*Downloader, *testutils.Memory, error) {
-	if err != nil {
-		t.Errorf("Failed to read the testcases:%s", err)
+	tests, err := loadTests("../testdata/online-organization-tests.json")
+	suite.NoError(err, "Failed to read the online tests")
+	downloader, err := NewDownloader(oauth2.NewClient(
+		context.TODO(),
+		oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: os.Getenv("GITHUB_TOKEN")},
+		)), suite.db)
+	suite.NoError(err, "Failed to init the downloader")
+	suite.downloader = downloader
+	for _, test := range tests.OrganizationsTests {
+		test := test
+		t.Run(fmt.Sprintf("Org %s with DB", test.Org), func(t *testing.T) {
+			testOrgWithDB(t, test, downloader, suite.db)
+		})
 	}
+}
 
-	downloader, storer, err := getDownloader()
-	require.NoError(t, err)
+func testRepoWithDB(t *testing.T, oracle testutils.RepositoryTest, d *Downloader, db *sql.DB) {
+	err := d.DownloadRepository(context.TODO(), oracle.Owner, oracle.Repository, oracle.Version)
+	require := require.New(t) // Make a new require object for the specified test, so no need to pass it around
+	require.Nil(err)
+	var (
+		htmlurl               string
+		createdAt             time.Time
+		private               bool
+		archived              bool
+		hasWiki               bool
+		topics                []string
+		numOfIssues           int
+		numOfIssueComments    int
+		numOfPRs              int
+		numOfPRReviewComments int
+	)
+	err = db.QueryRow("select htmlurl, created_at, private, archived, has_wiki, topics from repositories where owner_login = $1 and name = $2", oracle.Owner, oracle.Repository).Scan(&htmlurl, &createdAt, &private, &archived, &hasWiki, pq.Array(&topics))
+	require.NoError(err, "Error in retrieving repo")
+	err = db.QueryRow("select count(*) from issues where repository_owner = $1 and repository_name = $2", oracle.Owner, oracle.Repository).Scan(&numOfIssues)
+	require.NoError(err, "Error in retrieving issues")
+	err = db.QueryRow("select count(*) from issue_comments where repository_owner = $1 and repository_name = $2", oracle.Owner, oracle.Repository).Scan(&numOfIssueComments)
+	require.NoError(err, "Error in retrieving issue comments")
+	err = db.QueryRow("select count(*) from pull_requests where repository_owner = $1 and repository_name = $2", oracle.Owner, oracle.Repository).Scan(&numOfPRs)
+	require.NoError(err, "Error in retrieving pull requests")
+	err = db.QueryRow("select count(*) from pull_request_comments where repository_owner = $1 and repository_name = $2", oracle.Owner, oracle.Repository).Scan(&numOfPRReviewComments)
+	require.NoError(err, "Error in retrieving pull request comments")
+	require.Equal(oracle.URL, htmlurl)
+	require.Equal(oracle.CreatedAt, createdAt.String())
+	require.Equal(oracle.IsPrivate, private)
+	require.Equal(oracle.IsArchived, archived)
+	require.Equal(oracle.HasWiki, hasWiki)
+	require.ElementsMatch(oracle.Topics, topics)
+	require.Equal(numOfIssues, oracle.NumOfIssues, "Issues")
+	require.Equal(oracle.NumOfPRs, numOfPRs, "PRs")
```

Comment:
maybe we can refactor this part somehow? split queries with assertion to functions
it's not necessary but will not look so massive.

Reply:
Good idea